docs:updated readme with typesript support (#320)
moltin/js-sdk: README.md

})

``` 
### TypeScript Support

> The Moltin SDK is fully supported in Typescript. Each service is wrapped in a namespace which exposes all the necessary interfaces and endpoints that can be consumed.

Reviewer: ```suggestion
The Elastic Path Commerce Cloud JavaScript SDK is fully supported in Typescript. Each service is wrapped in a namespace which exposes all the necessary interfaces and endpoints that can be consumed.
```

> Each response will likely be returned as either a `Response` (single object returned) or a `ResponsePage` (an array of items) - see `src/types/core.d.ts` for more info.

Reviewer: ```suggestion
Each response will be returned as either a `Response` (single object returned) or a `ResponsePage` (an array of items) - see `src/types/core.d.ts` for more info.
```

Author: @shaunmaharaj Only reason I put likely was because of things like auth tokens and stuff, which are not returned. Might change it to, `the majority of responses will be returned`

> The majority of responses will be returned as either a `Response` (single object returned) or a `ResponsePage` (an array of items) - see `src/types/core.d.ts` for more info.

Reviewer: ```suggestion
Responses will be returned as either a `Response` (single object returned) or a `ResponsePage` (an array of items) - see `src/types/core.d.ts` for more info. Additional responses may be returned in a native object structure.
```

> Current exposed namespaces include `address`, `brand`, `cart`, `category`, `collectoin`, `core`, `currencies`, `customer`, `field`, `flow`, `inergration`, `inventory`, `job`, `order`, `product` and `transaction`. For further information on any of these services, please refer to the [docs](https://documentation.elasticpath.com/commerce-cloud/docs/api).

Reviewer 2: `collectoin`

Reviewer: `inergration`

Reviewer: While we're updating, change `docs` to `documentation`

> Current exposed namespaces include `address`, `brand`, `cart`, `category`, `collection`, `core`, `currencies`, `customer`, `field`, `flow`, `inergration`, `inventory`, `job`, `order`, `product` and `transaction`. For further information on any of these services, please refer to the [docs](https://documentation.elasticpath.com/commerce-cloud/docs/api).

Reviewer: ```suggestion
Current exposed namespaces include `address`, `brand`, `cart`, `category`, `collection`, `core`, `currencies`, `customer`, `field`, `flow`, `integration`, `inventory`, `job`, `order`, `product` and `transaction`. For further information on any of these services, please refer to the [documentation](https://documentation.elasticpath.com/commerce-cloud/docs/api).
```

        commodity_type: "physical",
        price: [
            {
                amount: 5499,
                currency: "USD",
                includes_tax: true
            }
        ]
    };
    
    const nP: Resource<Product> = await g.Products.Create(newProduct);
}
```

You can also extend any base interface compatible with flows to create any custom interfaces that you might be using.

```
interface CustomProduct extends product.ProductBase {
    customFlowField?: string
}

const newProduct: CustomProduct = {
    type: "product",
    name: "My Product",
    slug: "my-prod",
    sku: "my-prod",
    customFlowField: "My custome flow..."
    manage_stock: false,
    description: "Some description",
    status: "draft",
    commodity_type: "physical",
    price: [
        {
            amount: 5499,
            currency: "USD",
            includes_tax: true
        }
    ]
};
``` 

You can cast these custom interfaces on any of the endpoints, to ensure you expose the custom fields. 

```
const customProduct: Resource<CustomProduct> = await g.Products.Create<CustomProduct>(newProduct);

console.log(`New Flow Product`, customProduct.data.customWeight); // this will pass!
```

> You can also consume `filters`, `sorts` and `includes` from relevant services, to ensure you are querying the api correctly.

Reviewer: ```suggestion
You can also consume `filters`, `sorts` and `includes` from relevant services, to ensure you are querying the API correctly.
```

eg: 

```

> const filter: priduct.ProductFilter  = {

Reviewer 2: `priduct`
